src/cascadia/TerminalControl/TSFInputControl.cpp

```diff
@@ -0,0 +1,379 @@
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TSFInputControl::NotifyFocusEnter()
+    {
+        if (_editContext != nullptr)
+        {
+            _editContext.NotifyFocusEnter();
+        }
+    }
+
+    // Method Description:
+    // - NotifyFocusEnter handler for notifying CoreEditTextContext of focus leaving.
+    //   when TerminalControl no longer has focus.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TSFInputControl::NotifyFocusLeave()
+    {
+        if (_editContext != nullptr)
+        {
+            //_editContext.NotifyFocusLeave();
+        }
+    }
+
+    // Method Description:
+    // - Scales a Rect based on a scale factor
+    // Arguments:
+    // - rect: Rect to scale by scale
+    // - scale: amount to scale rect by
+    // Return Value:
+    // - Rect scaled by scale
+    inline Rect ScaleRect(Rect rect, double scale)
+    {
+        const float scaleLocal = gsl::narrow<float>(scale);
+        rect.X *= scaleLocal;
+        rect.Y *= scaleLocal;
+        rect.Width *= scaleLocal;
+        rect.Height *= scaleLocal;
+        return rect;
+    }
+
+    // Method Description:
+    // - Converts a COLORREF to Color
+    // Arguments:
+    // - colorref: COLORREF to convert to Color
+    // Return Value:
+    // - Color containing the RGB values from colorref
+    inline winrt::Windows::UI::Color ColorRefToColor(const COLORREF& colorref)
+    {
+        winrt::Windows::UI::Color color;
+        color.R = GetRValue(colorref);
+        color.G = GetGValue(colorref);
+        color.B = GetBValue(colorref);
+        return color;
+    }
+
+    // Method Description:
+    // - Handler for LayoutRequested event by CoreEditContext responsible
+    //   for returning the current position the IME should be placed
+    //   in screen coordinates on the screen.  TSFInputControls internal
+    //   XAML controls (TextBlock/Canvas) are also positioned and updated.
+    //   NOTE: documentation says application should handle this event
+    // Arguments:
+    // - sender: CoreTextEditContext sending the request.
+    // - args: CoreTextLayoutRequestedEventArgs to be updated with position information.
+    // Return Value:
+    // - <none>
+    void TSFInputControl::_layoutRequestedHandler(CoreTextEditContext sender, CoreTextLayoutRequestedEventArgs const& args)
+    {
+        auto request = args.Request();
+
+        // Get window in screen coordinates, this is the entire window including tabs
+        auto windowBounds = Window::Current().CoreWindow().Bounds();
+
+        // Get the cursor position in text buffer position
+        auto cursorArgs = winrt::make_self<CursorPositionEventArgs>();
+        _currentCursorPositionHandlers(*this, *cursorArgs);
+        COORD cursorPos = { gsl::narrow_cast<SHORT>(cursorArgs->CurrentPosition().X), gsl::narrow_cast<SHORT>(cursorArgs->CurrentPosition().Y) }; //_terminal->GetCursorPosition();

```

Comment:
```suggestion
        const COORD cursorPos = { gsl::narrow_cast<SHORT>(cursorArgs->CurrentPosition().X), gsl::narrow_cast<SHORT>(cursorArgs->CurrentPosition().Y) }; //_terminal->GetCursorPosition();
```

This looks const as you only use it as an in parameter below.

Reply:
Resolved